File: src/266_palindrome_permutation/palindrome_permutation.py
class Solution:
    def canPermutePalindrome(self, s: str) -> bool:
        # TODO: Implement the solution.
        return False


def run_test(s: str, expected: bool) -> None:
    actual = Solution().canPermutePalindrome(s)
    status = "PASS" if actual == expected else "FAIL"
    print(f"{actual} | {expected} | {status}")


if __name__ == "__main__":
    run_test("code", False)
    run_test("aab", True)
    run_test("carerac", True)
    run_test("a", True)
    run_test("abc", False)

File: src/266_palindrome_permutation/palindrome_permutation.cpp
#include <iostream>
#include <string>
using namespace std;

class Solution {
   public:
    bool canPermutePalindrome(string s) {
        // TODO: Implement the solution.
        return false;
    }
};

void runTest(string s, bool expected) {
    Solution solution;
    bool actual = solution.canPermutePalindrome(s);
    string status = actual == expected ? "PASS" : "FAIL";
    cout << boolalpha << actual << " | " << expected << " | " << status << endl;
}

int main() {
    runTest("code", false);
    runTest("aab", true);
    runTest("carerac", true);
    runTest("a", true);
    runTest("abc", false);

    return 0;
}
